client/checkout/api/index.js

```diff
@@ -444,6 +444,45 @@ export default class WCPayAPI {
 			} );
 	}
 
+	/**
+	 * Confirm Stripe payment with fallback for rate limit error.
+	 *
+	 * @param {Object} confirmParams Confirm payment request parameters.
+	 * @param {Object|StripeElements} elements Stripe elements.
+	 * @param {string|null} paymentIntentSecret Payment intent secret used to validate payment on rate limit error
+	 *
+	 * @return {Promise} The final promise for the request to the server.
+	 */
+	async handlePaymentConfirmation(
+		confirmParams,
+		elements,
+		paymentIntentSecret
+	) {
+		const stripe = this.getStripe();
+		const confirmPaymentResult = await stripe.confirmPayment( {
+			elements,
+			confirmParams,
+		} );
+		if (
+			paymentIntentSecret &&
+			confirmPaymentResult.error &&
+			'lock_timeout' === confirmPaymentResult.error.code
+		) {
+			const paymentIntentResult = await stripe.retrievePaymentIntent(
+				paymentIntentSecret
+			);
+			if (
+				! paymentIntentResult.error &&
+				'succeeded' === paymentIntentResult.paymentIntent.status
```

Comment:
Do we also need to check against `requires_capture` and `processing`?

See in our gateway code: https://github.com/Automattic/woocommerce-payments/blob/ca4325592ba30bf2d15bd09c5d4e407e2a747fc2/includes/class-wc-payment-gateway-wcpay.php#L65

Reply:
Based on the conversation with Stripe, their suggestion was to redirect when succeeded status is set.

Reply:
Okay, but is that because we told them we wanted to do that in our message or is this because this is truly the only status to check? I wouldn't be so sure.

Why wouldn't we want to redirect if the payment intent's status is `requires_capture`? There's no sense of payment confirmed in that case since the merchant explicitly wants to capture the payment manually later, so it seems right also to redirect.

For the `processing` status, I'm not sure when this status is set like that however..

What do es@Automattic/transact-architecture think of this?

Reply:
I will merge this PR and open a new issue to address this topic.